float64 X_length
float64 Y_length
float64 YAW_length